libcudacxx/include/cuda/std/__cccl/cuda_toolkit.h

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_CUDA_TOOLKIT_H
+#define __CCCL_CUDA_TOOLKIT_H
+
+#include <cuda/std/__cccl/compiler.h>
+#include <cuda/std/__cccl/system_header.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if _CCCL_CUDA_COMPILATION() || _CCCL_HAS_INCLUDE(<cuda_runtime_api.h>)
+#  define _CCCL_HAS_CTK() 1
+#else // ^^^ has cuda toolkit ^^^ / vvv no cuda toolkit vvv
+#  define _CCCL_HAS_CTK() 0
+#endif // ^^^ no cuda toolkit ^^^
+
+// CUDA compilers preinclude cuda_runtime.h, so we need to include it here to get the CUDART_VERSION macro
+#if _CCCL_HAS_CTK() && !_CCCL_CUDA_COMPILATION()
+#  include <cuda_runtime_api.h>
+#endif // _CCCL_HAS_CTK() && !_CCCL_CUDA_COMPILATION()
+
+#if _CCCL_CUDA_COMPILATION() && _CCCL_CUDACC_BELOW(CUDART_VERSION / 1000, (CUDART_VERSION % 1000) / 10)
+#  error \
+    "CUDA compiler must come from a CUDA toolkit version greater than or equal to the included CUDA Runtime version"
+#endif // _CCCL_CUDA_COMPILATION() && _CCCL_CUDACC_BELOW(CUDART_VERSION / 1000, (CUDART_VERSION % 1000) / 10)
```

Comment:
Sorry, I totally forgot to consider nvc++, which can include older toolkits, but has a different versioning scheme.

@dkolsen-pgi do you have any requirements regarding the combination of nvc++ and cuda headers?

Reply:
NVC++ predefines the macro `CUDA_VERSION` to match the version of the CUDA Toolkit that the NVC++ driver thinks it is using.  The macro `CUDART_VERSION` is defined by the toolkit headers.  It would be good to check that those two macros are compatible even when NVC++ is the CUDA compiler.  That will catch problems that might arise when the user messes with the include path.